source/addonHandler/__init__.py

```diff
@@ -388,16 +389,16 @@ def enable(self, shouldEnable):
 			if self.name in state["pendingEnableSet"]:
 				# Undoing a pending enable.
 				state["pendingEnableSet"].discard(self.name)
-			# No need to disable an addon that is already disabled.
+			# No need to disable an addon that is already disabled, except if all add-ons are disabled.
 			# This also prevents the status in the add-ons dialog from saying "disabled, pending disable"
-			elif self.name not in state["disabledAddons"]:
+			elif self.name not in state["disabledAddons"] or globalVars.appArgs.disableAddons:
 				state["pendingDisableSet"].add(self.name)
 		# Record enable/disable flags as a way of preparing for disaster such as sudden NVDA crash.
 		saveState()
 
 	@property
 	def isRunning(self):

```

Comment:
Rather than changing this property, could you change isDisabled otherwise? Or will this lead to issues I'm not aware of?
What it will do, is log every add-on that is disabled. I consider this more of a feature than a bug, though.

Reply:
That's what I initially did, but @feerrenrut told me to modify `isRunning` instead.

Reply:
Is disabled relates to the users saved state. `isRunning` does not, instead `isRunning` is the negation of several other reasons it might not be running: eg blocked, disabled, pending install

Reply:
Ah, that's fine then.